exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -73,14 +60,29 @@ func (amp *AzureMachinePool) SetDefaultSSHPublicKey() error {
 }
 
 // SetIdentityDefaults sets the defaults for VMSS Identity.
-func (amp *AzureMachinePool) SetIdentityDefaults(subscriptionID string) {
+func (amp *AzureMachinePool) SetIdentityDefaults(client client.Client) error {
 	// Ensure the deprecated fields and new fields are not populated simultaneously
 	if amp.Spec.RoleAssignmentName != "" && amp.Spec.SystemAssignedIdentityRole != nil && amp.Spec.SystemAssignedIdentityRole.Name != "" {
 		// Both the deprecated and the new fields are both set, return without changes
 		// and reject the request in the validating webhook which runs later.
-		return
+		return nil
 	}
 	if amp.Spec.Identity == infrav1.VMIdentitySystemAssigned {
+		machinePool, err := azureutil.FindParentMachinePoolWithRetry(amp.Name, client, 5)
+		if err != nil {
+			return errors.Wrap(err, "failed to find parent machine pool")
```

Comment:
We're returning the error from the helper function but appending that error in `SetDefaults`. I think that's what we want because if this part of the function fails there's nothing else we can do to set identity defaults, but we do want to attempt to set other defaults in the AzureMachinePool spec.